jax/_src/lax/parallel.py

```diff
@@ -1113,11 +1125,14 @@ def _pdot_impl(x, y, *, axis_name, pos_contract, pos_batch):
 
 @pdot_p.def_abstract_eval
 def _pdot_abstract_eval(x, y, *, axis_name, pos_contract, pos_batch):
-  # TODO: avals with names, check inputs are mapped along axis_name, eliminate
   if not len(set(axis_name)) == len(axis_name): raise ValueError
-  return lax.dot_general_p.abstract_eval(
+  pos_aval = lax.dot_general_p.abstract_eval(
       x, y, dimension_numbers=[pos_contract, pos_batch],
       precision=None, preferred_element_type=None)
+  named_shape = {name: size
+                 for aval in (x, y) for name, size in aval.named_shape.items()
+                 if name not in axis_name}
```

Comment:
In line with the comments above, we want to check here that the inputs indeed have the names present to be contracted over, right? (As per the TODO that was deleted!)

Since this PR doesn't yet add tests for `pdot`, I suggest keeping some form of TODO here, mentioning that it's untested logic.